### ISSUE TITLE
Support complete lifecycle of a device - fortimanager_dvm_device

### ISSUE THREAD

**Issue by jack1902** [fortinetdev/terraform-provider-fortimanager#31]
It is possible to add a device and delete a device via seperate resource blocks within this provider which feels quite anti-terraform.

Terraform supports `CRUD` which can be achieved for a `device` by  making use of the following:
- C - https://github.com/fortinetdev/terraform-provider-fortimanager/blob/9fe6eef7aa466aeda62e5d2dbefbd5eb341f2690/fmg/resource_dvm_cmd_add_device.go#L153
- R - This doesn't appear to be implemented anywhere in particular. However this bit here looks like a READ so unsure why this wasn't implemented https://github.com/fortinetdev/terraform-provider-fortimanager/blob/9fe6eef7aa466aeda62e5d2dbefbd5eb341f2690/fmg/resource_dvm_cmd_update_device.go#L66
- U  - Both of the following appear to be the same
  - https://github.com/fortinetdev/terraform-provider-fortimanager/blob/9fe6eef7aa466aeda62e5d2dbefbd5eb341f2690/fmg/resource_dvm_cmd_add_device.go#L153 
  - https://github.com/fortinetdev/terraform-provider-fortimanager/blob/9fe6eef7aa466aeda62e5d2dbefbd5eb341f2690/fmg/resource_dvm_cmd_update_device.go#L57
- D - https://github.com/fortinetdev/terraform-provider-fortimanager/blob/9fe6eef7aa466aeda62e5d2dbefbd5eb341f2690/fmg/resource_dvm_cmd_del_device.go#L57

I feel like it would be very beneficial to use terraform to manage the full lifecycle of a device within fortimanager as terraform is already being used to provision virtual fortigates. Previously looked at `fortiOS` provider but due to versioning of the fortimanager we are now using this provider instead however so far it hasn't felt like a proper terraform provider as it appears to use seperate resource snippets for each of the CRUD operations that terraform is meant to abstract away from you.

**Comment by DARB-CCM-S-20** [comment]
Agreed, in my example I am using the `fortimanager_dvm_cmd_add_device`. I would expect that performing a `tf destroy` on this resource would remove the device from the fortimanager. However, this isn't the case and it seems to do this I instead need to `tf apply` on `fortimanager_dvm_cmd_del_device` which is rather counterintuitive. Furthermore, this means this resource must be in a separate configuration. So in order to tear down an environment you must `tf apply` on one configuration then tf destroy` on another.

**Comment by MaxxLiu22** [comment]
Hi @jack1902 @DARB-CCM-S-20 ,

Thank you for raising this FMG flaw, I have reported it to the development team. They provide the information that FMG system use separate exec API to achieve CRUD operations to manager devices instead of one API method with all operations (add update get delete), and during adding a device FMG uses another task API to get device adding process status and result instead of just return 200OK, so terraform managing device resource is kind of different from other resources, but they will improve this feature in the future. Thanks again for your suggestion, that is helpful for us to improve our products.

Thanks,
Maxx

**Comment by mohamed-ldgm** [comment]
Same request, if this could be improved it would be great. The provider is barely usable because of this. It is still valuable :)
